File: 594-longest-harmonious-subsequence/longest-harmonious-subsequence.cpp
class Solution {
public:
    int findLHS(vector<int>& nums) {
        unordered_map<int, int>mp;

        for(int i=0;i<nums.size();i++){
            mp[nums[i]]++;
        }

        int maxLen = 0;
        

        for(auto [key, value]: mp){

            if(mp.count(key+1)){
                maxLen = max(maxLen , value + mp[key + 1]);
            }

        }

        return maxLen;
    }
};

File: 594-longest-harmonious-subsequence/longest-harmonious-subsequence.ts
function findLHS(nums: number[]): number {
        const freq: Map<number, number> = new Map();
    
    for (const num of nums) {
        freq.set(num, (freq.get(num) || 0) + 1);
    }

    let maxLen: number = 0;

    for (const [key, value] of freq.entries()) {
        if (freq.has(key + 1)) {
            maxLen = Math.max(maxLen, value + freq.get(key + 1)!);
        }
    }

    return maxLen;
};